### ISSUE TITLE
Selecting a (real-time) segment on a dashboard with many widgets seems to trigger the (partially same) archiving process several times

### ISSUE THREAD

**Issue by peterbo** [matomo-org/matomo#15926]
Reproducible by creating a dashboard with 10+ widgets and building a segment (e.g. Entry-page == XYZ AND Channel Type == ZXY).

In the database, you can see a lot of queries that are exactly the same and taking up a lot of CPU time while basically querying / sorting the same data:

![1b](https://user-images.githubusercontent.com/1018973/81459635-516bc180-91a1-11ea-827e-301e56d33454.jpg)
![2b](https://user-images.githubusercontent.com/1018973/81459637-529cee80-91a1-11ea-8338-df00337e22be.jpg)
![3b](https://user-images.githubusercontent.com/1018973/81459638-53358500-91a1-11ea-8c2b-558e6377f82c.jpg)
![4b](https://user-images.githubusercontent.com/1018973/81459639-53ce1b80-91a1-11ea-9fb1-d8542c92f6b8.jpg)
![5b](https://user-images.githubusercontent.com/1018973/81459643-5466b200-91a1-11ea-9566-5724c8529e64.jpg)

The given instance is quite small, but real-time segment calculation is very slow. Fixing this could speed up real-time segment archiving a lot.

An example for one of those queries running simultaneously:
`INSERT INTO logtmpsegmentb4ca4bef0234682057a466a33290929d (idvisit) /* idSegments = [83] */

			SELECT
				distinct log_visit.idvisit as idvisit
			FROM
				log_visit AS log_visit
			WHERE
				( log_visit.visit_last_action_time >= '2020-03-31 22:00:00'
				AND log_visit.visit_last_action_time <= '2020-04-30 21:59:59'
				AND log_visit.idsite IN ('195') )
                AND
                ( log_visit.referer_type = '6' AND log_visit.visit_entry_idaction_url = '209938' )
			ORDER BY
				log_visit.idvisit ASC

INSERT INTO logtmpsegmentb4ca4bef0234682057a466a33290929d (idvisit) /* idSegments = [83] */

			SELECT
				distinct log_visit.idvisit as idvisit
			FROM
				log_visit AS log_visit
			WHERE
				( log_visit.visit_last_action_time >= '2020-03-31 22:00:00'
				AND log_visit.visit_last_action_time <= '2020-04-30 21:59:59'
				AND log_visit.idsite IN ('195') )
                AND
                ( log_visit.referer_type = '6' AND log_visit.visit_entry_idaction_url = '209938' )
			ORDER BY
				log_visit.idvisit ASC`

**Comment by tsteur** [comment]
@peterbo  I reckon this is a duplicate of https://github.com/matomo-org/matomo/issues/8444 ?

**Comment by peterbo** [comment]
Hey @tsteur! The commenters are talking about different things, but Matt's comment is exactly it:

>if one loads dashboard, and 10 widgets call the API at the same time (same second), we expect that Piwik will trigger archiving only once (in one thread) and other threads will wait for this process, then when archiving is finished all widgets loading will be unblocked and will read this same archive

**Comment by tsteur** [comment]
Cheers, will mark it as a duplicate for now.
